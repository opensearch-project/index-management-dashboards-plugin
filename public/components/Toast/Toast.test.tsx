/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import { waitFor } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import { SimpleEuiToast } from "./index";

describe("SimpleEuiToast show", () => {
  it("render the component", async () => {
    await act(async () => {
      SimpleEuiToast.addSuccess("Success information");
    });
    expect(document.body).toMatchSnapshot();
    expect(document.querySelector('[data-test-subj="toast_Success information"]')).not.toBeNull();
    await act(async () => {
      SimpleEuiToast.addDanger("Error information");
    });
    expect(document.querySelector('[data-test-subj="toast_Error information"]')).not.toBeNull();
    await act(async () => {
      SimpleEuiToast.show({
        toastLifeTimeMs: 10,
        title: "Test quick destroy",
      });
    });
    await waitFor(() => {
      expect(document.querySelector('[data-test-subj="toast_Test quick destroy"]')).toBeNull();
    });
  });
});
