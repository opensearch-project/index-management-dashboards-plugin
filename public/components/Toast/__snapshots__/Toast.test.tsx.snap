// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleEuiToast show render the component 1`] = `
<body>
  <div
    data-role="SimpleEuiToast"
    id="EUI_SIMPLE_TOAST_MOUNT_ID"
  >
    <div
      aria-live="polite"
      class="euiGlobalToastList euiGlobalToastList--right"
      role="region"
    >
      <div
        class="euiToast euiToast--success euiGlobalToastListItem"
        data-test-subj="toast_Success information"
        id="toast_0"
      >
        <p
          class="euiScreenReaderOnly"
        >
          A new notification appears
        </p>
        <div
          aria-label="Notification"
          class="euiToastHeader"
          data-test-subj="euiToastHeader"
        >
          <span
            class="euiToastHeader__title"
          >
            Success information
          </span>
        </div>
        <button
          aria-label="Dismiss toast"
          class="euiToast__closeButton"
          data-test-subj="toastCloseButton"
          type="button"
        >
          EuiIconMock
        </button>
      </div>
    </div>
  </div>
</body>
`;
